src/generic/datavgen.cpp

```diff
@@ -2046,69 +2073,225 @@ bool wxDataViewMainWindow::EnableDropTarget( const wxDataFormat &format )
+    }
+    else
+    {
+        wxDataViewTreeNode* node = GetTreeNodeByRow(row);
+        if (!node)
+            return dropItemInfo;
+
+        dropItemInfo.m_item = node->GetItem();
+
+        unsigned int itemStart = GetLineStart(row);
+        int itemHeight = GetLineHeight(row);
+
+        bool insertAbove = yy - itemStart < itemHeight*0.15;       // 15%, TODO: configurable
+        if (insertAbove)
+        {
+            // Can be treated as 'insertBelow" with the small difference:
+            node = GetTreeNodeByRow(row - 1);   // We need the node from the previous row
+
+            dropItemInfo.m_hint = DropHint_Above;
+
+            if (!node)
+            {
+                // Seems to be dropped in the root node
+
+                dropItemInfo.m_indentLevel        = 0;
+                dropItemInfo.m_proposedDropIndex  = 0;
+                dropItemInfo.m_item               = wxDataViewItem();
+
+                return dropItemInfo;
+            }
+
+        }
+
+        bool insertBelow = yy - itemStart > itemHeight*0.85;        // 15%, TODO: configurable
+        if (insertBelow)
+            dropItemInfo.m_hint = DropHint_Below;
+
+        if (insertBelow || insertAbove)
+        {
+            // Insert inside the 'item' or after (below) it. Depends on:
+            // 1 - the 'item' is a container
+            // 2 - item's child index in it's parent (the last in the parent or not)
+            // 3 - expanded or not
+            // 4 - mouse x position
+
+            unsigned int start_x = 0;       // Expander column x position start
+            wxDataViewColumn* const expander = GetExpanderColumnOrFirstOne(GetOwner());
+            for (unsigned int i = 0; i < GetOwner()->GetColumnCount(); i++)
+            {
+                wxDataViewColumn* col = GetOwner()->GetColumnAt(i);
+                if (col->IsHidden())
+                    continue;   // skip it!
+
+                if (col == expander)
+                    break;
+
+                start_x += col->GetWidth();
+            }
+
+            const int expanderWidth = wxRendererNative::Get().GetExpanderSize(this).GetWidth();
+
+            int level = node->GetIndentLevel();
+
+            wxDataViewTreeNode* prevAscendNode = node;
+            wxDataViewTreeNode* ascendNode = node;
+            while (ascendNode != NULL)
+            {
+                dropItemInfo.m_indentLevel = level + 1;
+
+                if (m_owner->GetModel()->IsContainer(ascendNode->GetItem()))
+                {
+                    // Item can be inserted
+
+                    int itemPosition = ascendNode->FindChildByItem(prevAscendNode->GetItem());
+                    dropItemInfo.m_proposedDropIndex  = itemPosition == wxNOT_FOUND ? 0 : itemPosition + 1;
+                    dropItemInfo.m_item               = ascendNode->GetItem();
+
+                    if (IsExpanded(row) && itemPosition != ascendNode->GetChildNodes().size() - 1)
```

Comment:
This line also triggers a signed/unsigned warning, but here I'm actually not sure if it's harmless, i.e. I think there could be an actual bug here: what happens when child nodes is empty? The RHS is `UINT_MAX` then, does this comparison do the right thing? Or is it impossible for the size to be 0 here?

Also, should we use `itemPosition` or `m_proposedDropIndex` in this comparison? I'm not really sure what is it doing, a comment here would probably be helpful.

Reply:
Thanks for pointing out this!
It's really bug here.